cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/client/AzureClient.java

```diff
@@ -1006,6 +1007,20 @@ public void grantKeyVaultAccessPolicyToServicePrincipal(String resourceGroupName
         });
     }
 
+    public boolean checkKeyVaultAccessPolicyForServicePrincipal(String resourceGroupName, String vaultName, String principalObjectId) {
+        return handleAuthException(() -> {
+            List<AccessPolicy> accessPolicies = azure.vaults()
+                    .getByResourceGroup(resourceGroupName, vaultName)
+                    .accessPolicies();
+            for (int i = accessPolicies.size() - 1; i >= 0; i--) {
+                if (accessPolicies.get(i).inner().objectId().equals(principalObjectId)) {
+                    return true;
+                }
+            }
+            return false;
```

Comment:
The old-fashioned looping certainly works, but mostly makes sense when one is interested in the item index where the match is found. Otherwise, the Java 8 Streams API provides a shorter and more elegant solution:
```
return accessPolicies.stream().anyMatch(policy -> principalObjectId.equals(policy.inner().objectId()));
```
Note: I swapped the target & arg of `equals()`, so that no NPE will be encountered even if some policy entry has a `null` `objectId`. (Not really likely, but better be safe.)